### PR TITLE
feat(generator): add options to disable client class

### LINKED ISSUE #5935: generator: add option to exclude Client class generation
In order to better support services in which we only want to generate the stack from the Connection down, provide a mechanism to invoke the microgenerator such that no Client classes are emitted.

Alternately, we could continue to have the microgenerator emit the Client files for such services and just ignore them.

### PR DESCRIPTION
**Pull request by coryan**
Fixes #5935

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7833)

### CODE REVIEW

File: generator/internal/descriptor_utils.cc [hunk 1]
```diff
@@ -641,12 +641,18 @@ std::vector<std::unique_ptr<GeneratorInterface>> MakeGenerators(
     std::vector<std::pair<std::string, std::string>> const& vars) {
   std::vector<std::unique_ptr<GeneratorInterface>> code_generators;
   VarsDictionary service_vars = CreateServiceVars(*service, vars);
+  bool omit_client = false;
+  for (auto const& v : vars) {
```
Comment (devbww) [inline generator/internal/descriptor_utils.cc:645]: Should you look in `service_vars` instead (and then use `find()`)?

File: generator/standalone_main.cc [hunk 2]
```diff
@@ -209,6 +209,9 @@ int main(int argc, char** argv) {
     for (auto const& retry_code : service.retryable_status_codes()) {
       args.emplace_back("--cpp_codegen_opt=retry_status_code=" + retry_code);
     }
+    if (service.omit_client()) {
+      args.emplace_back("--cpp_codegen_opt=omit_client=true");
```
Comment (devbww) [inline generator/standalone_main.cc:213]: Opt: You may want to add a case to `generator/internal/codegen_utils_test.cc`.

File: generator/internal/descriptor_utils.cc (hunk 1 above)
Comment (coryan) [reply generator/internal/descriptor_utils.cc]: Done, for whatever reason my previous attempt at this failed.

File: generator/standalone_main.cc (hunk 2 above)
Comment (coryan) [reply generator/standalone_main.cc:213]: Done.

**Review by devbww — APPROVED** [review]

**Pull request merged**